pkgs/os-specific/darwin/apple-sdk/sdkRoot.nix

```diff
@@ -41,28 +41,24 @@ runCommand "sdkroot-${sdkVersion}" { } ''
 
   install -D '${../../../build-support/setup-hooks/role.bash}' "$out/nix-support/setup-hook"
   cat >> "$out/nix-support/setup-hook" <<-hook
-  #
-  # See comments in cc-wrapper's setup hook. This works exactly the same way.
-  #
-  [[ -z \''${strictDeps-} ]] || (( "\$hostOffset" < 0 )) || return 0
-
   sdkRootHook() {
     # See ../../../build-support/setup-hooks/role.bash
     local role_post
     getHostRoleEnvHook
 
     # Only set the SDK root if one has not been set via this hook or some other means.
-    if [[ ! \$NIX_CFLAGS_COMPILE =~ isysroot ]]; then
+    local cflagsVar=NIX_CFLAGS_COMPILE\''${role_post}
+    if [[ ! \''${!cflagsVar} =~ isysroot ]]; then
       export NIX_CFLAGS_COMPILE\''${role_post}+=' -isysroot $out/${sdkName}.sdk'
     fi
   }
 
   # See ../../../build-support/setup-hooks/role.bash
   getTargetRole
 
-  addEnvHooks "\$targetOffset" sdkRootHook
+  addEnvHooks "\$hostOffset" sdkRootHook
```

Comment:
This means we run the hooks in the context of the host rather than the target? Because the SDK root is necessary during the build rather than at run-time?

Reply:
I think so? The SDK root is used by the host compiler for the target system.